wdl-lint/src/rules/matching_parameter_meta.rs

```diff
@@ -104,26 +127,41 @@ impl Rule for MatchingParameterMetaRule {
     }
 }
 
-/// Checks for both missing and extra items in a `parameter_meta` section.
+/// Checks for both missing and extra items in a `parameter_meta` section
+/// along with the order of the items.
 fn check_parameter_meta(
     parent: &SectionParent,
-    expected: impl Iterator<Item = (Ident, Span)>,
+    expected: Vec<Decl>,
     param_meta: ParameterMetadataSection,
     diagnostics: &mut Diagnostics,
     exceptable_nodes: &Option<&'static [SyntaxKind]>,
 ) {
-    let expected: HashMap<_, _> = expected.map(|(i, s)| (i.text().to_string(), s)).collect();
-
-    let actual: HashMap<_, _> = param_meta
+    let expected_map: HashMap<_, _> = expected
+        .iter()
+        .map(|decl| (decl.name().text().to_string(), decl.name().span()))
+        .collect();
+    let actual_map: HashMap<_, _> = param_meta
         .items()
         .map(|m| {
             let name = m.name();
             (name.text().to_string(), name.span())
         })
         .collect();
 
-    for (name, span) in &expected {
-        if !actual.contains_key(name) {
+    // We only consider the intersection of the expected and actual keys
```

Comment:
This is a change to this rule correct? And it is now out-of-sync with the function documentation. This rule should check for extraneous and missing entries in `parameter_meta`.